sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -421,68 +427,77 @@ export class MessageSender extends LinkEntity {
   }
 
   /**
-   * Initializes the sender session on the connection.
+   * Initializes the `ClientEntityContext.sender` associated with this
+   * `MessageSender`.
+   *
+   * If the connection is already open this method resolves immediately.
    */
-  private async _init(options?: SenderOptions): Promise<void> {
-    try {
-      // isOpen isConnecting  Should establish
-      // true     false          No
-      // true     true           No
-      // false    true           No
-      // false    false          Yes
-      if (!this.isOpen()) {
-        log.error(
-          "[%s] The sender '%s' with address '%s' is not open and is not currently " +
-            "establishing itself. Hence let's try to connect.",
-          this._context.namespace.connectionId,
-          this.name,
-          this.address
-        );
-        this.isConnecting = true;
-        await this._negotiateClaim();
-        log.error(
-          "[%s] Trying to create sender '%s'...",
-          this._context.namespace.connectionId,
-          this.name
-        );
-        if (!options) {
-          options = this._createSenderOptions({});
+  public async open(options?: SenderOptions): Promise<void> {
+    if (this.isOpen()) {
+      return;
+    }
+
+    return defaultLock.acquire(this.openLock, async () => {
```

Comment:
We have now lost the log for acquiring lock that we had before

```
log.sender(	
          "Acquiring lock %s for initializing the session, sender and " +	
            "possibly the connection.",	
          this.senderLock	
        );
```

Reply:
retry() wasn't even logging before despite it potentially taking the most monstrous lock. So that might help with future diagnostics as well.

Reply:
Done.